src/mongocxx/client.hpp

```diff
@@ -61,7 +68,7 @@ class MONGOCXX_API client {
     ///   Additional options that cannot be specified via the mongodb_uri
     ///
     client(
-        const class uri& mongodb_uri = mongocxx::uri(),
+        const class uri& mongodb_uri,
```

Comment:
I don't feel strongly either way, but I think it's all right to force the caller to be deliberate about creating an empty URI.  As you mentioned it isn't normal behavior to do this, so it's fine to discourage doing it with the extra typing.